src/arrayMethodReduce.js

```diff
@@ -5,7 +5,25 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const thisArray = [...this];
+
+    if (thisArray.length === 0 && initialValue === undefined) {
+      throw new TypeError('reduce of empty array with no initial value');
+    }
```

Comment:
Great solution! It's easy and simple to understand!) But there is one moment. There won't be an error in the reduce method if the array is empty and the initial value is undefined. So you don't need this part in your solution I guess.

Reply:
Actually, It will be the error in such case, so good job👍